clients/fossid-webapp/src/main/kotlin/model/result/Snippet.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (See <https://github.com/oss-review-toolkit/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.fossid.model.result
+
+data class Snippet(
```

Comment:
Are there some docs available for this class and its properties? Is really almost everything in this class optional, i.e. you cannot rely on the presence of properties?

Reply:
No unfortunately as always with FossID, there is no documentation of the model available.

Maybe we can make the assumption `id`, `matchType`, `fieIdId` and `scanId` are not null, for the rest I don't know.

Please view https://github.com/oss-review-toolkit/ort/blob/34ca6c65da51415bfdf09c06223867fb140d2d32/clients/fossid-webapp/src/test/assets/return-type/mappings/apiphp-list_scan_snippets.json for example of the response.

Reply:
> Maybe we can make the assumption `id`, `matchType`, `fieIdId` and `scanId` are not null

Yes, please make as many properties non-nullable as you confidently can, to simplify any upcoming code that makes use of these properties.

Reply:
Ok I queried the snippets of several files of the Semver4j project and I make some changes to the class.